NitroxPatcher/Patches/Dynamic/ConstructorInput_OnCraftingBegin_Patch.cs

```diff
@@ -28,15 +28,24 @@ public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnum
                 if (instruction.opcode.Equals(INJECTION_OPCODE) && instruction.operand.Equals(INJECTION_OPERAND))
                 {
                     /*
-                     * TransientLocalObjectManager.Add(TransientLocalObjectManager.TransientObjectType.CONSTRUCTOR_INPUT_CRAFTED_GAMEOBJECT, gameObject);
+                     * Callback(constructor, gameObject, techType, duration);
                      */
-                    yield return new CodeInstruction(OpCodes.Ldc_I4_0);
+                    yield return original.Ldloc<ConstructorInput>(0);
                     yield return original.Ldloc<GameObject>(0);
-                    yield return new CodeInstruction(OpCodes.Call, Reflect.Method(() => Add(default(TransientObjectType), default(object))));
+                    yield return new CodeInstruction(OpCodes.Ldarg_0);
+                    yield return new CodeInstruction(OpCodes.Ldfld, TARGET_METHOD.DeclaringType.GetField("techType", BindingFlags.Instance | BindingFlags.Public));
+                    yield return new CodeInstruction(OpCodes.Ldarg_0);
+                    yield return new CodeInstruction(OpCodes.Ldfld, TARGET_METHOD.DeclaringType.GetField("duration", BindingFlags.Instance | BindingFlags.Public));
```

Comment:
You can use Reflect.Property((ConstructorInput c) => c.TechType), we're no longer using raw reflection

Will also work on private/protected attributes since Nitrox is now using Publicized assemblies

Reply:
For clarity, the TARGET_METHOD.DeclaringType is not ConstructorInput, it is the class automatically generated for the IEnumerable.  I don't think there is any way to access it via the anonymous function route.